core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-feedback-thread-modal.controller.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for ImprovementFeedbackThreadModal.
+ */
+
+angular.module('oppia').controller('ImprovementFeedbackThreadModalController', [
+  '$scope', '$uibModalInstance', 'AlertsService', 'DateTimeFormatService',
+  'EditabilityService', 'ThreadDataService', 'ThreadStatusDisplayService',
+  'UserService', 'thread',
+  function(
+      $scope, $uibModalInstance, AlertsService, DateTimeFormatService,
+      EditabilityService, ThreadDataService, ThreadStatusDisplayService,
+      UserService, thread) {
+    UserService.getUserInfoAsync().then(function(userInfo) {
+      $scope.isUserLoggedIn = userInfo.isLoggedIn();
+    });
+    $scope.activeThread = thread;
+    $scope.STATUS_CHOICES = ThreadStatusDisplayService.STATUS_CHOICES;
+    $scope.getLabelClass = ThreadStatusDisplayService.getLabelClass;
+    $scope.getHumanReadableStatus = (
+      ThreadStatusDisplayService.getHumanReadableStatus);
+    $scope.getLocaleAbbreviatedDatetimeString = (
+      DateTimeFormatService.getLocaleAbbreviatedDatetimeString);
+    $scope.EditabilityService = EditabilityService;
+
+    // Initial load of the thread list on page load.
+    $scope.tmpMessage = {
+      status: $scope.activeThread.status,
+      text: '',
+    };
+
+    $scope.getTitle = function() {
+      return $scope.activeThread.subject;
+    };
+
+    // TODO(Allan): Implement ability to edit suggestions before
+    // applying.
+    $scope.addNewMessage = function(threadId, tmpText, tmpStatus) {
```

Comment:
Do we need to take in a `threadId` here? It's always equal to `thread.threadId`, right?

Reply:
Done.

Reply:
@brianrodri `addNewMessage` needs to have `threadId` param, the e2e tests were failing when I removed it.

Reply:
```typescript
$scope.addNewMessage = function(tmpText, tmpStatus) {
  // ...
  ThreadDataService.addNewMessage(
    thread.threadId, tmpText, tmpStatus, ...);
}

Reply:
I've tried this way now It's working!